metric_providers/disk/io/cgroup/container/source.c

```diff
@@ -0,0 +1,189 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <unistd.h>
+#include <sys/time.h>
+#include <string.h> // for strtok
+#include <getopt.h>
+
+typedef struct container_t { // struct is a specification and this static makes no sense here
+    char path[BUFSIZ];
+    char *id;
+} container_t;
+
+typedef struct disk_io_t { // struct is a specification and this static makes no sense here
+    long int rbytes;
+    long int wbytes;
+} disk_io_t;
+
+
+// All variables are made static, because we believe that this will
+// keep them local in scope to the file and not make them persist in state
+// between Threads.
+// in any case, none of these variables should change between threads
+static int user_id = 0;
```

Comment:
I would probably move this into the main as if the `getuid` is somehow deleted you are assuming root. Or set it to -1 so the program fails

Reply:
Fair point. We are never using the user_id though to check for user id but only to construct a path in the sysfs.

I changed it in all the files though as I believe if we use it in the future then you are right and it can be a security bug.